demux/demux_mkv.c

```diff
@@ -1743,15 +1743,18 @@ static int demux_mkv_open_audio(demuxer_t *demuxer, mkv_track_t *track)
     {
         track->parse = true;
     } else if (!strcmp(codec, "flac")) {
-        unsigned char *ptr = extradata;
-        unsigned int size = extradata_len;
-        if (size < 4 || ptr[0] != 'f' || ptr[1] != 'L' || ptr[2] != 'a'
-            || ptr[3] != 'C') {
-            extradata = talloc_size(sh_a, 4);
-            extradata_len = 4;
-            memcpy(extradata, "fLaC", 4);
-        }
         parse_flac_chmap(&sh_a->channels, extradata, extradata_len);
+
+        if (extradata_len < 8) {
+            extradata = NULL;
+            extradata_len = 0;
+        } else {
+            // ffmpeg will generally discard the 'fLaC' header and store
+            // only the initial STREAMINFO after parsing the channel map,
+            // so let's do the same for compability (say, recorder.c muxing)
+            extradata += 8;
+            extradata_len = 34; // FLAC_STREAMINFO_SIZE
```

Comment:
This is not safe without checking how many bytes are actually present.
Why not `extradata_len -= 8;`?

Reply:
> Why not extradata_len -= 8;?

Because we have no need for the extended extradata, only the FLAC STREAMINFO is parsed. The remaining is usually just the channel mapping (which we/ffmpeg generate when muxing), and other(?) possible non-playback metadata.

> This is not safe

Fair, I'll go with a `8 + 34` in the if.

Reply:
Ah, so it would break if the rest of the data is kept?